### PR TITLE
DOE-2891 Push Version Tagged Docker Image

### PR DESCRIPTION
**Pull request by jeffmaher**
### JIRA Issue Link
https://osi-cwds.atlassian.net/browse/DOE-2891

### Technical Description
Pushes a version tagged Docker image at build time instead of the `latest` tag.

### Tests
N/A

### Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by jeffmaher** [comment]
Added a comment to the Jira ticket:
https://osi-cwds.atlassian.net/browse/DOE-2891?focusedCommentId=91452&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-91452

File: Jenkinsfile-complete.groovy [hunk 1]
```diff
@@ -83,7 +83,7 @@ node('tpt2-slave') {
         stage('Build Docker') {
             buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'createDockerImage'
             withDockerRegistry([credentialsId: '6ba8d05c-ca13-4818-8329-15d41a089ec0']) {
-                buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'pushDockerLatest -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
+                buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'pushDockerVersionTagged -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile-complete.groovy]: @jeffmaher  This is intentional. We are publishing tagged Docker Image on last stage of pipeline after all stages are successful.

Comment (LeonidMarushevskyi) [reply Jenkinsfile-complete.groovy]: Latest is publish just for deployment purpose.

Comment (jeffmaher) [reply Jenkinsfile-complete.groovy]: @LeonidMarushevskyi : Sounds good. I'll revert that. Can I also add in a `pushVersionTagged` call to run in parallel to this?

File: Jenkinsfile-complete.groovy [hunk 2]
```diff
@@ -85,6 +85,9 @@ node('tpt2-slave') {
             withDockerRegistry([credentialsId: '6ba8d05c-ca13-4818-8329-15d41a089ec0']) {
                 buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'pushDockerLatest -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
```
Comment (jeffmaher) [inline Jenkinsfile-complete.groovy]: Added this back in, then added version tagged push.

Comment (LeonidMarushevskyi) [reply Jenkinsfile-complete.groovy]: @jeffmaher we don't want to publish tagged image until the application was deployed and integration tests are successful. 

On the last stage, we have dockerPublish which published tagged image.

Comment (jeffmaher) [reply Jenkinsfile-complete.groovy]: @LeonidMarushevskyi : Thanks! I totally missed that. 

Perhaps the fix then is to move the GitHub tagging back to that point too. What are your thoughts on that?

**Comment by jeffmaher** [comment]
@LeonidMarushevskyi : Was looking over with @Srikanthchowdhary , and he's going to move the tagging near where the publishing of artifacts happens so that we don't have a tag that we don't want (i.e. before tests pass).

**Comment by LeonidMarushevskyi** [comment]
@jeffmaher good idea.

File: Jenkinsfile-complete.groovy [hunk 3]
```diff
@@ -85,11 +85,11 @@ node('tpt2-slave') {
...
             }
+            withDockerRegistry([credentialsId: '6ba8d05c-ca13-4818-8329-15d41a089ec0']) {
+                buildInfo = rtGradle.run buildFile: 'build.gradle', tasks: 'pushDockerVersionTagged -DRelease=$RELEASE_PROJECT -DBuildNumber=$BUILD_NUMBER -DCustomVersion=$OVERRIDE_VERSION'
```
Comment (LeonidMarushevskyi) [inline Jenkinsfile-complete.groovy]: Don't need to push tagged Docker Image.

**Pull request merged**